### PR TITLE
Updating/repairing the SDK lookup tests

### PR DESCRIPTION
**Pull request by johnbeisner**
1. Fixing the: 'SdkLookup_Must_Look_For_Available_Versions_Before_Looking_Into_Another_Folder' test to be deterministic.
2. Adding semantical version tests.

A follow-up PR to: https://github.com/dotnet/core-setup/pull/2055
https://github.com/dotnet/cli/issues/6339

@dotnet/dotnet-cli

### CODE REVIEW

File: test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs [hunk 1]
```diff
@@ -230,12 +232,117 @@ public void SdkLookup_Must_Look_For_Available_Versions_Before_Looking_Into_Anoth
...
                 .And
-                .HaveStdErrContaining(Path.Combine(_cwdSelectedMessage, "9999.0.0-global-dummy"));
+                .HaveStdErrContaining(Path.Combine(_cwdSelectedMessage, "9999.0.0-global-dummy", _dotnetSdkDllMessageTerminator));
 
             // Remove dummy folders from user dir
             DeleteAvailableSdkVersions(_userSdkBaseDir, "9999.0.0", "9999.0.0-dummy");
         }
 
+        [Fact]
+        public void SdkLookup_Must_Pick_The_Latest_Semantical_Version()
```
Comment (livarcocc) [inline test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: I wouldn't call it latest, I would call it highest.

Btw, good to see this test here.

Comment (eerhardt) [inline test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: (nit) `Semantic` Version.  Not `Semantical`.

Comment (johnbeisner) [inline test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: done...

File: test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs [hunk 2]
```diff
@@ -230,12 +232,117 @@ public void SdkLookup_Must_Look_For_Available_Versions_Before_Looking_Into_Anoth
...
                 .And
-                .HaveStdErrContaining(Path.Combine(_cwdSelectedMessage, "9999.0.0-global-dummy"));
+                .HaveStdErrContaining(Path.Combine(_cwdSelectedMessage, "9999.0.0-global-dummy", _dotnetSdkDllMessageTerminator));
 
             // Remove dummy folders from user dir
             DeleteAvailableSdkVersions(_userSdkBaseDir, "9999.0.0", "9999.0.0-dummy");
         }
 
+        [Fact]
+        public void SdkLookup_Must_Pick_The_Highest_Semantical_Version()
```
Comment (eerhardt) [inline test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: (nit) `Semantic` Version. Not `Semantical`.

Comment (johnbeisner) [inline test/HostActivationTests/GivenThatICareAboutMultilevelSDKLookup.cs]: yup, done...

**Pull request merged**